plugins/providers/bigquery/client.go

```diff
@@ -110,38 +110,21 @@ func (c *bigQueryClient) GetTables(ctx context.Context, datasetID string) ([]*Ta
 	return results, nil
 }
 
-func (c *bigQueryClient) ResolveDatasetRole(role string) (bq.AccessRole, error) {
-	switch role {
-	case DatasetRoleReader:
-		return bq.ReaderRole, nil
-	case DatasetRoleWriter:
-		return bq.WriterRole, nil
-	case DatasetRoleOwner:
-		return bq.OwnerRole, nil
-	default:
-		return "", ErrInvalidRole
-	}
-}
-
 func (c *bigQueryClient) GrantDatasetAccess(ctx context.Context, d *Dataset, user, role string) error {
 	dataset := c.client.Dataset(d.DatasetID)
 	metadata, err := dataset.Metadata(ctx)
 	if err != nil {
 		return err
 	}
 
-	bqRole, err := c.ResolveDatasetRole(role)
-	if err != nil {
-		return err
-	}
 	for _, a := range metadata.Access {
-		if a.Entity == user && a.Role == bqRole {
+		if a.Entity == user && a.Role == bq.AccessRole(role) {
 			return ErrPermissionAlreadyExists
 		}
 	}
 	update := bq.DatasetMetadataToUpdate{
 		Access: append(metadata.Access, &bq.AccessEntry{
-			Role:       bqRole,
+			Role:       bq.AccessRole(role),
```

Comment:
please make sure granting a role such as `roles/bigquery.dataViewer` (one example returned from queryGrantableRoles) instead of `READER` is possible using this API, because as per the client library codes, `bq.AccessRole` are only READER, WRITER, and OWNER

Reply:
@rahmatrhd here we are adding all possible roles on BigQuery and allowed roles on datasets READER, WRITER, and OWNER
`var roles = []bq.AccessRole{bq.OwnerRole, bq.WriterRole, bq.ReaderRole}`

here we are adding these default roles to roles to not missed these roles

Reply:
handle the scenario